backend/src/main/java/com/bakdata/conquery/models/index/MapIndexService.java

```diff
@@ -0,0 +1,98 @@
+package com.bakdata.conquery.models.index;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import com.bakdata.conquery.io.jackson.Injectable;
+import com.bakdata.conquery.io.jackson.MutableInjectableValues;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.univocity.parsers.common.record.Record;
+import com.univocity.parsers.csv.CsvParser;
+import com.univocity.parsers.csv.CsvParserSettings;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.text.StringSubstitutor;
+import org.jetbrains.annotations.NotNull;
+
+@RequiredArgsConstructor
+@Slf4j
+public class MapIndexService implements Injectable {
+
+	private final CsvParserSettings csvParserSettings;
+
+	// TODO provide a way to evict mappings
+	private final LoadingCache<Key, Map<String, String>> mappings = CacheBuilder.newBuilder().build(new CacheLoader<Key, Map<String, String>>() {
+		@Override
+		public Map<String, String> load(@NotNull Key key) throws Exception {
+			log.info("Started to parse mapping {}", key);
+
+			final Map<String, String> int2ext = new HashMap<>();
+
+			final CsvParser csvParser = new CsvParser(csvParserSettings);
+
+			csvParser.beginParsing(key.csv.openStream());
+			Record row;
+			for (row = csvParser.parseNextRecord(); row != null; row = csvParser.parseNextRecord()) {
+				final StringSubstitutor substitutor = new StringSubstitutor(row::getString, "{{", "}}", StringSubstitutor.DEFAULT_ESCAPE);
+
+				substitutor.setEnableUndefinedVariableException(true);
+				final String internalValue = row.getString(key.internalColumn);
+
+				if (internalValue == null) {
+					return null;
+				}
+
+				try {
+					final String externalValue = substitutor.replace(key.externalTemplate);
+					int2ext.put(internalValue, externalValue);
+				}
+				catch (IllegalArgumentException exception) {
+					log.warn("Missing template values for line {}. Internal value: {}", csvParser.getContext()
+																								 .currentLine(), internalValue, (Exception) (log.isTraceEnabled()
+																																			 ? exception
+																																			 : null));
+				}
+
+			}
+			csvParser.stopParsing();
+
+			log.info("Finished parsing mapping {} with {} entries", key, int2ext.size());
+			return int2ext;
+		}
+	});
+
+
+	public CompletableFuture<Map<String, String>> getMapping(URL csv, String internalColumn, String externalTemplate) {
+		// TODO may use a specific executor service here
+		return CompletableFuture.supplyAsync(() -> {
+			try {
+				return mappings.get(new Key(csv, internalColumn, externalTemplate));
```

Comment:
Du machst jetzt zwar nachgelagerte Deduplizierung, es wird aber immer noch in den Konzepten reichlich Duplikate geben, deswegen mein Vorschlag die Mappings als separates Namespaced Objekt / Json einzuführen, um sie dann in den Selects und Templates referenzieren zu können.

Reply:
Momentan kann es nicht zu einer Duplizierung kommen, wenn die selben Dinge im Mapping stehen (`csv, internalColumn, externalTemplate`). Deswegen gibt es keine Deduplizierung. 

Dein Vorschlag die Mappings zu referenzieren wollte ich nachgelagert machen.

Reply:
>Momentan kann es nicht zu einer Duplizierung kommen

Glaube wir meinen das gleiche aber reden aneinander vorbei: Wenn ich in zwei Konzepten das gleiche Mapping benutzen will muss ich es 2 mal definieren eben pro Select. Das ist doppelte Arbeit, die ich vermeiden wollte mit meinem Vorschlag unten. Dass es intern dedupliziert wird habe ich verstanden.

Reply:
Ja du hast recht hier braucht man Referenzen, da es sogar leicht vorkommen, kann dass man das selbe Mapping in einem Konzept braucht, weil es z.B ein first und und last select hat. -> Extra PR